### DISCUSSION TITLE
umi + qiankun本地开发的时候，如何让源码不转义，devtool无效

### DISCUSSION THREAD

**Discussion by senfish** [umijs/umi#11697] [Q&A] [answered]
现象：`umi`  + `qiankun` 再本地开发的时候，调试子应用代码的时候，发现所有的子应用代码都被转义了，导致调试难度很大。如图
![image](https://github.com/umijs/umi/assets/49187663/090d668e-bace-4142-9ead-8c64e7153175)

`.umirc.ts` 里面`devtool` 的配置都是 `devtool: process.env.NODE_ENV === 'development' ? 'eval-cheap-module-source-map' : 'source-map'` 
`umi`版本为`3.5.20` 

期望： 想让devtool生效

**Comment by fz6m** [comment, marked as answer]
umi 3 不维护了，建议升级到 umi 4
